.github/workflows/check.yml

```diff
@@ -4,23 +4,23 @@ jobs:
   check:
     strategy:
       matrix:
-        os:
-          [
-            '{"runs-on": "ubuntu-latest"}',
-            '{"runs-on": "macos-latest"}',
-            '{"runs-on": "ubuntu-latest", "container": "ubuntu:18.04"}',
-            '{"runs-on": "windows-latest"}',
-          ]
-    runs-on: ${{ fromJSON(matrix.os).runs-on }}
-    container: ${{ fromJSON(matrix.os).container }}
+        system:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+        include:
+          - system: ubuntu-18.04
+            container: ubuntu:18.04    
+    runs-on: ${{ matrix.system }}
+    container: ${{ matrix.container }}
```

Comment:
does this not work?

```suggestion
          - ubuntu-18.04
    runs-on: ${{ matrix.system }}
```

Reply:
was the intent:

- test on both latest (i.e. 20.04+) and 18.04, or
- test `container:`

Reply:
I think that it was testing the `container:` option, because plain `ubuntu` containers don't contain many of the popular packages shipped with the official GitHub Actions runners.

Reply:
Keeping the 18.04 version number was just a hack to keep the matrix engine from merging it with the `ubuntu-latest` variant.